### PR TITLE
add structure molecular surface visual

### PR DESCRIPTION
**Pull request by arose**
- Add "Structure Molecular Surface" visual

### CODE REVIEW

File: src/mol-repr/structure/visual/util/common.ts [hunk 1]
```diff
@@ -213,43 +214,67 @@ export function getUnitConformationAndRadius(structure: Structure, unit: Unit, s
...
+                _ys.push(_y);
+                _zs.push(_z);
                 l.element = eI;
                 _rs.push(sizeTheme.size(l));
-                _id.push(m + j);
+
+                if (differentRoot) {
+                    const idx = childUnit ? SortedArray.indexOf(childUnit.elements, eI) : -1;
+                    if (idx === -1) {
+                        _id.push(-2);
```
Comment (dsehnal) [inline src/mol-repr/structure/visual/util/common.ts:266]: `-2`?

File: src/mol-repr/structure/visual/util/common.ts [hunk 2]
```diff
@@ -213,43 +214,67 @@ export function getUnitConformationAndRadius(structure: Structure, unit: Unit, s
...
+                    }
+                } else {
+                    _id.push(getSerialIndex(unit, eI));
+                }
             }
-            m += elements.length;
         }
         xs = _xs, ys = _ys, zs = _zs, rs = _rs;
         id = _id;
+        indices = OrderedSet.ofRange(0, id.length);
```
Comment (dsehnal) [inline src/mol-repr/structure/visual/util/common.ts:277]: Is this correct with the negative indices in the ids?

File: src/mol-repr/structure/visual/util/common.ts (hunk 1 above)
Comment (arose) [reply src/mol-repr/structure/visual/util/common.ts]: this is to mark for filtering/ignoring in CPU MC, added a comment

File: src/mol-repr/structure/visual/util/common.ts (hunk 2 above)
Comment (arose) [reply src/mol-repr/structure/visual/util/common.ts:277]: yes

**Pull request merged**